### PR TITLE
[Homework #19] - React Hooks by Oleg Gorodenskyi

### PR DESCRIPTION
**Pull request by OlegGorod**
@ilya-ischenko Hi Ilya. Check my hw, plz.

### CODE REVIEW

File: homeworks/oleg-gorodenskyi/19-react-hooks/src/App.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      localStorage.setItem(id, JSON.stringify(inputValue));
+      setToDoList([...toDoList, { value: inputValue, id }]);
+      setInputValue('');
+    }
+  }
+  useEffect(() => {
+    const storedData = Object.entries(localStorage).map(([key, data]) => ({
+      id: key,
+      value: JSON.parse(data),
+    }));
```
Comment (ilya-ischenko) [inline homeworks/oleg-gorodenskyi/19-react-hooks/src/App.js:32]: What is it? Why are you trying to map entire local storage? You should save your tasks to local storage and get this tasks from localstorage. In your case you map the entire local storage where you can have a lot of data
https://javascript.info/localstorage

File: homeworks/oleg-gorodenskyi/19-react-hooks/src/App.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import ToDoItem from './components/ToDoItem';
+
+function App() {
+  const [toDoList, setToDoList] = useState([]);
+  const [inputValue, setInputValue] = useState('');
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    const valueToStore = JSON.stringify(inputValue);
+    const isValueExist = Object.values(localStorage).includes(valueToStore);
```
Comment (ilya-ischenko) [inline homeworks/oleg-gorodenskyi/19-react-hooks/src/App.js:14]: What is it? Why are you trying to map entire local storage? You should save your tasks to local storage and get this tasks from localstorage. In your case you map the entire local storage where you can have a lot of data
https://javascript.info/localstorage

File: homeworks/oleg-gorodenskyi/19-react-hooks/src/App.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    if (isValueExist) {
+      setInputValue('');
+      alert('This value is already exist');
+    } else if (inputValue.length > 25) {
+      setInputValue('');
+      alert('The value should be not longer 25 symbols');
+    } else {
+      const id = v4();
+      localStorage.setItem(id, JSON.stringify(inputValue));
```
Comment (ilya-ischenko) [inline homeworks/oleg-gorodenskyi/19-react-hooks/src/App.js:23]: You just writing id and input value to local storage, but how are you going to distinguish this value (whether it's task or not) when you get it from local storage?

File: homeworks/oleg-gorodenskyi/19-react-hooks/src/App.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+  return (
+    <div className="App">
+      <form onSubmit={handleSubmit}>
+        <input
+          type="text"
+          placeholder='Create To-Do Task'
+          value={inputValue}
+          onChange={e => setInputValue(e.target.value)}
+        />
```
Comment (ilya-ischenko) [inline homeworks/oleg-gorodenskyi/19-react-hooks/src/App.js:52]: Input should be separate component

File: homeworks/oleg-gorodenskyi/19-react-hooks/src/App.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        <button className='submit' type="submit">Add</button>
```
Comment (ilya-ischenko) [inline homeworks/oleg-gorodenskyi/19-react-hooks/src/App.js:53]: Button should be separate component

File: homeworks/oleg-gorodenskyi/19-react-hooks/src/components/ToDoItem.js [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+      )}
+      {!isEdit && (
+        <button className="editBtn" onClick={() => setIsEdit(!isEdit)}>
+          <img src={editIcon} alt="edit" />
+        </button>
+      )}
+      <button className="deleteBtn" onClick={() => deleteToDo(toDo.id)}>
+        <img src={deleteIcon} alt="delete" />
+      </button>
+    </div>
```
Comment (ilya-ischenko) [inline homeworks/oleg-gorodenskyi/19-react-hooks/src/components/ToDoItem.js:78]: All buttons/inputs should be separate components

**Review by ilya-ischenko — COMMENTED** [review]
In overall I could not fully test your HW because it crashed on my PC. It's because of you read the entire local storage

File: homeworks/oleg-gorodenskyi/19-react-hooks/src/App.js (hunk 1 above)
Comment (OlegGorod) [reply homeworks/oleg-gorodenskyi/19-react-hooks/src/App.js:32]: OMG, yeah. I didn't work with a LS for a long time and forgot about it. lol)

**Review by ilya-ischenko — APPROVED** [review]
Well done!
Approved.

**Pull request merged**